src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,114 @@
 package seedu.address.logic.parser;
 
+import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_CODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
 
 import java.util.Arrays;
+import java.util.stream.Stream;
 
 import seedu.address.logic.commands.FindCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.assignment.DeadlineContainsKeywordsPredicate;
+import seedu.address.model.assignment.ModuleCodeContainsKeywordsPredicate;
 import seedu.address.model.assignment.NameContainsKeywordsPredicate;
+import seedu.address.model.assignment.PriorityContainsKeywordsPredicate;
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    private static boolean isDateFormat(String keyword) {
+        return keyword.matches("\\d{2}-\\d{2}-\\d{4}");
```

Comment:
What if the user types in 90-90-9090?

Reply:
parseDeadline method (implemented in the parse method) will throw a parseException error so this isDateFormat only ensures that the user keys in something of this format

Reply:
WOW AMENDED thanks for eyeballing